test/DualTest.jl

```diff
@@ -504,4 +505,13 @@ end
     @test !isfinite(dinf)
 end
 
+if VERSION >= v"1.6.0-rc1"
+    @testset "@printf" begin
+        for T in (Float16, Float32, Float64, BigFloat)
+            d1 = Dual(one(T))
+            @test_nowarn @printf("Testing @printf: %.2e\n", d1)
```

Comment:
Would it make more sense to use `@sprintf` and check the result?

Reply:
Thanks, that's a good suggestion! I added an additional test using `@sprintf` where the resulting string is checked.